builder/mixconfig.go

```diff
@@ -0,0 +1,194 @@
+				{"VERSIONS_PATH", &config.VersionDir},
+				{"YUM_CONF", &config.YumConf},
+			}},
+		{
+			section: "swupd",
+			pairs: []configPair{
+				{"BUNDLE", &config.Bundle},
+				{"CONTENTURL", &config.ContentURL},
+				{"FORMAT", &config.Format},
+				{"VERSIONURL", &config.VersionURL},
+			}},
+		{
+			section: "Server",
+			pairs: []configPair{
+				{"debuginfo_banned", &config.DebugInfoBanned},
+				{"debuginfo_lib", &config.DebugInfoLib},
+				{"debuginfo_src", &config.DebugInfoSrc},
+			}},
+		{
+			section: "Mixer",
+			pairs: []configPair{
+				{"LOCAL_BUNDLE_DIR", &config.LocalBundleDir},
+				{"LOCAL_REPO_DIR", &config.RepoDir},
+				{"LOCAL_RPM_DIR", &config.RPMDir},
+			}},
+	}
+}
+
+func (config *MixConfig) mapToIni() (*ini.File, error) {
+	cfg := ini.Empty()
+
+	for _, entry := range config.getMapping() {
+		section, err := cfg.NewSection(entry.section)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, pair := range entry.pairs {
+			if _, err = section.NewKey(pair.key, *pair.value); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return cfg, nil
+}
+
+func (config *MixConfig) mapFromIni(cfg *ini.File) {
+	for _, entry := range config.getMapping() {
+		section, err := cfg.GetSection(entry.section)
+		if err != nil {
+			continue
+		}
+
+		for _, pair := range entry.pairs {
+			key, err := section.GetKey(pair.key)
+			if err == nil {
+				*pair.value = key.String()
+			}
+		}
+	}
+}
+
+// LoadDefaults sets sane defaults for all the config values in MixCOnfig
+func (config *MixConfig) LoadDefaults() error {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return nil
+	}
+
+	// [Builder]
+	config.BundleDir = filepath.Join(pwd, "mix-bundles")
+	config.Cert = filepath.Join(pwd, "Swupd_Root.pem")
+	config.StateDir = filepath.Join(pwd, "update")
+	config.VersionDir = pwd
+	config.YumConf = filepath.Join(pwd, ".yum-mix.conf")
+
+	// [Swupd]
+	config.Bundle = "os-core-update"
+	config.ContentURL = "<URL where the content will be hosted>"
+	config.Format = "1"
+	config.VersionURL = "<URL where the version of the mix will be hosted>"
+
+	// [Server]
+	config.DebugInfoBanned = "true"
+	config.DebugInfoLib = "/usr/lib/debug"
+	config.DebugInfoSrc = "/usr/src/debug"
+
+	// [Mixer]
+	config.LocalBundleDir = filepath.Join(pwd, "local-bundles")
+	config.RPMDir = ""
+	config.RepoDir = ""
+
+	return nil
+}
+
+// CreateDefaultConfig creates a default builder.conf using the active
+// directory as base path for the variables values.
+func (config *MixConfig) CreateDefaultConfig(localrpms bool) error {
+	if err := config.LoadDefaults(); err != nil {
+		return err
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	builderconf := filepath.Join(pwd, "builder.conf")
+
+	if localrpms {
+		config.RPMDir = filepath.Join(pwd, "local-rpms")
+		config.RepoDir = filepath.Join(pwd, "local-yum")
+	}
+
+	fmt.Println("Creating new builder.conf configuration file...")
+
+	cfg, err := config.mapToIni()
```

Comment:
Prefer a template approach to generate the config for the following reasons:

(a) we can opt to not add some values (or add them commented), to let the user take the default
(b) we can add some other comments into the configuration file (like explain sections and how certain variables are used).

Reply:
> Prefer a template approach to generate the config

While I agree with both of your reasons when it comes to creating the default config file, I strongly want the ability to write the config _from a data structure_.  Ultimately, I want the ability to move the Mix Bundles List out of `mixbundles` and into `builder.conf`, and that means "editing the Mix Bundles List" means editing `builder.conf`.

Reply:
I don't see why those things couldn't live together. We can have a template and keep the ability of writing to the config file. The only downside that I can see now is that with the current lib, any comments on the config file would be gone once we edit it.

Reply:
> I strongly want the ability to write the config from a data structure.

The suggestion won't prevent that. My point was more: for the initial mix, it make more sense to use the template approach. The template can take a config object and run with it, or a function can take a config object and use Fprintfs.

Note about using library facilities for writing: there will be challenges if we want to keeping user comments (and in general don't cause disruptive changes to the file), so unless library was made thinking about that case, we'll have to handle this ourselves.